comp/metadata/inventorychecks/inventorychecksimpl/inventorychecks.go

```diff
@@ -197,9 +208,41 @@ func (ic *inventorychecksImpl) getPayload() marshaler.JSONMarshaler {
 		}
 	}
 
+	logsMetadata := make(map[string][]metadata)
+	if sources, isSet := ic.sources.Get(); isSet {
+		for _, logSource := range sources.GetSources() {
+			if _, found := logsMetadata[logSource.Name]; !found {
+				logsMetadata[logSource.Name] = []metadata{}
+			}
+
+			parsedJSON, err := cjson.Marshal(logSource.Config)
+			if err != nil {
+				ic.log.Debugf("could not parse log configuration for source metadata %s: %v", logSource.Name, err)
+				continue
+			}
+
+			// Compute log source status
+			logSourceStatus := "pending"
+			if logSource.Status.IsSuccess() {
+				logSourceStatus = "success"
+			} else if logSource.Status.IsError() {
+				logSourceStatus = "error"
+			}
+
+			logsMetadata[logSource.Name] = append(logsMetadata[logSource.Name], metadata{
+				"config": string(parsedJSON),
+				"state": map[string]string{
+					"error":  logSource.Status.GetError(),
+					"status": logSourceStatus,
+				},
+			})
+		}
+	}
+
 	return &Payload{
-		Hostname:  ic.hostname,
-		Timestamp: time.Now().UnixNano(),
-		Metadata:  payloadData,
+		Hostname:     ic.hostname,
+		Timestamp:    time.Now().UnixNano(),
+		Metadata:     payloadData,
+		LogsMetadata: logsMetadata,
```

Comment:
I might be wrong, but I'm not seeing any test checking the new `LogsMetadata` information. If it is true that that we are not checking those in the tests, could we please add some tests 😄